File: results/java_codex/FIND_THE_LARGEST_SUBARRAY_WITH_0_SUM_1.java
int maxLen(int arr[], int n)
{
    // Creates an empty hashMap hM
    unordered_map<int, int> hM;

    int sum = 0;      // Initialize sum of elements
    int max_len = 0;  // Initialize result

    // Traverse through the given array
    for (int i = 0; i < n; i++)
    {
        // Add current element to sum
        sum += arr[i];

        if (arr[i] == 0 && max_len == 0)
            max_len = 1;

        if (sum == 0)
            max_len = i+1;

        // Look this sum in hash table
        if (hM.find(sum) != hM.end())
            max_len = max(max_len, i - hM[sum]);
        else // Else put this sum in hash table
            hM[sum] = i;
    }

    return max_len;
}

File: TransCoder_tokenized_test_set_functions/python/FIND_THE_LARGEST_SUBARRAY_WITH_0_SUM_1.py
def maxLen ( arr ) :
    hash_map = {
    }
    
    max_len = 0
    curr_sum = 0
    for i in range ( len ( arr ) ) :
        curr_sum += arr [ i ]
        if arr [ i ] is 0 and max_len is 0 :
            max_len = 1
        if curr_sum is 0 :
            max_len = i + 1
        if curr_sum in hash_map :
            max_len = max ( max_len, i - hash_map [ curr_sum ] )
        else :
            hash_map [ curr_sum ] = i
    return max_len
